db/db_flush_test.cc

```diff
@@ -1106,6 +1106,193 @@ TEST_F(DBFlushTest, MemPurgeAndCompactionFilter) {
   ASSERT_EQ(Get(KEY5), p_v5);
 }
 
+TEST_F(DBFlushTest, MemPurgeWALSupport) {
+  Options options = CurrentOptions();
+
+  options.statistics = CreateDBStatistics();
+  options.statistics->set_stats_level(StatsLevel::kAll);
+  options.create_if_missing = true;
+  options.compression = kNoCompression;
+  options.inplace_update_support = false;
+  options.allow_concurrent_memtable_write = true;
+
+  // Enforce size of a single MemTable to 64MB (64MB = 67108864 bytes).
+  options.write_buffer_size = 1 << 20;
```

Comment:
It would be nice not to use such large sizes on things, for test efficiency.

Also there's a lot of repetitive / copy-paste in this test that would be nice to trim down if reasonably possible.

Reply:
+1 on the repetitive/CTRL-C - CTRL-V code, I will try and make this unit test more human readable.